### PR TITLE
Release v3.1.1

### PR DESCRIPTION
**Pull request by PawelLipski**

### CODE REVIEW

File: CHANGE-NOTES.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 
 ## Unreleased
```
Comment (PawelLipski) [inline CHANGE-NOTES.md:3]: `v3.1.1`

File: frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java [hunk 2]
```diff
@@ -80,6 +87,40 @@ private void processMacheteGeneratedEntry(MacheteGeneratedEntry macheteEntry, An
...
+    }
+  }
+
+  @UIEffect
+  private void saveDocumentBeforeCheck(PsiFile file) {
+    val fileDocManager = FileDocumentManager.getInstance();
+    fileDocManager.saveDocument(Objects.requireNonNull(fileDocManager.getDocument(file.getVirtualFile())));
+  }
+
+  private boolean isBranchNameRepeated(String fileContent, String branchName) {
```
Comment (PawelLipski) [inline frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/highlighting/MacheteAnnotator.java:118]: To be reworked within https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1272

File: CHANGE-NOTES.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
...
+- Removed automatic rediscovery in case of empty machete file
```
Comment (mkondratek) [inline CHANGE-NOTES.md:4]: missing mention about repeated entry fix

**Review by mkondratek — CHANGES_REQUESTED** [review]

File: CHANGE-NOTES.md (hunk 3 above)
Comment (PawelLipski) [reply CHANGE-NOTES.md:4]: If you could piggyback that onto one of your PRs

Comment (mkondratek) [reply CHANGE-NOTES.md:4]: https://github.com/VirtusLab/git-machete-intellij-plugin/pull/1282

**Pull request merged**